JsonLogic/Rule.cs

```diff
@@ -136,7 +146,7 @@ public override void Write(Utf8JsonWriter writer, Rule value, JsonSerializerOpti
 	{
 		if (value.Source != null)
 		{
-			JsonSerializer.Serialize(writer, value.Source, options);
+			JsonSerializer.Serialize(writer, value.Source, LogicSerializerContext.Default.JsonNode);
```

Comment:
I don't know the context that these JsonConverters are being used, but I think we want a pattern in the Write methods that's similar to the read -- that is, respecting the full TypeInfoResolver chain that's in the JsonSerializerOptions passed in.

By switching from "options" to the JsonSerializerContext from our module only, we prevent a caller from extending the list.

As we discussed in Slack, I think we should have a Write<T> overload that can help validate but that still passes the options along for any nested deserialization.